20230301/mike.java

```diff
@@ -0,0 +1,44 @@
+/*
+To practice different algos, I use quicksort to complete this question.
+Althrough quicksort may exceed O(nlogn), it pass all of the test case on LeetCode.
+Maybe we should try heapsort and counting sort next time!
+*/
+
+class Solution {
+    public int[] sortArray(int[] nums) {
+        quickSort(nums, 0, nums.length-1);
+        
+        return nums;
+    }
+    
+    public void quickSort(int[] nums, int start, int end){
+        if(start >= end) {
+          return;  
+        }
+        
+        int pivot = nums[(start + end)/2];
+        int left = start;
+        int right = end;
+        
+        while (left <= right) {
+            while (left <= right && nums[left] < pivot ) {
+                left++;
+            }
+        
+            while (left <= right && nums[right] > pivot) {
+                right--;
+            }
+            
+            if(left <= right) {
+                int temp = nums[left];
+                nums[left] = nums[right];
+                nums[right] = temp;
```

Comment:
it seems like we can't swap two elements with a one-liner like python. 😈 
```python
nums[left], nums[right] = nums[right], nums[left]
```

this is what chatgpt suggests in addition to the traditional `temp` manner.
```java
// Create an example array
int[] arr = { 1, 2, 3, 4, 5 };

// Swap elements at positions 1 and 3 without temporary variable
arr[1] = arr[1] ^ arr[3];
arr[3] = arr[1] ^ arr[3];
arr[1] = arr[1] ^ arr[3];
```